tests/validation/cattlevalidationtest/core/test_rancher_compose_commands.py

```diff
@@ -0,0 +1,620 @@
+
+    container_list1 = get_service_container_list(super_client, service1)
+    assert len(container_list1) == service1.scale
+    for container in container_list1:
+        assert container.state == "running"
+
+    container_list2 = get_service_container_list(super_client, service2)
+    assert len(container_list2) == service2.scale
+    for container in container_list2:
+        assert container.state == "running"
+
+    delete_all(client, [env])
+
+
+@if_compose_data_files
+def test_rancher_compose_services_delete(super_client, client,
+                                         rancher_compose_container):
+
+    env_name = random_str().replace("-", "")
+
+    # Create an environment using up
+    launch_rancher_compose_from_file(
+        client, RCCOMMANDS_SUBDIR, "dc1.yml", env_name,
+        "up -d", "Starting project", "rc1.yml")
+    env, service = get_env_service_by_name(client, env_name, "test1")
+
+    # Confirm service is active and the containers are running
+    assert service.state == "active"
+    check_config_for_service(super_client, service, {"test1": "value1"}, 1)
+
+    container_list = get_service_container_list(super_client, service)
+    assert len(container_list) == service.scale
+    for container in container_list:
+        assert container.state == "running"
+
+    launch_rancher_compose_from_file(
+        client, RCCOMMANDS_SUBDIR, "dc1.yml", env_name,
+        "rm -f", "Deleting", "rc1.yml")
+
+    time.sleep(10)
+    # Confirm service is active and the containers are running
+    service = client.wait_success(service, 300)
+    assert service.state == "removed"
+
+    container_list = get_service_container_list(super_client, service)
+    assert len(container_list) == 0
+
+    delete_all(client, [env])
+
+
+@if_compose_data_files
+def test_rancher_compose_services_scale(super_client, client,
+                                        rancher_compose_container):
+
+    env_name = random_str().replace("-", "")
+
+    # Create an environment using up
+    launch_rancher_compose_from_file(
+        client, RCCOMMANDS_SUBDIR, "dc1.yml", env_name,
+        "up -d", "Starting project", "rc1.yml")
+    env, service = get_env_service_by_name(client, env_name, "test1")
+
+    # Confirm service is active and the containers are running
+    assert service.state == "active"
+    check_config_for_service(super_client, service, {"test1": "value1"}, 1)
+
+    container_list = get_service_container_list(super_client, service)
+    assert len(container_list) == service.scale
+    for container in container_list:
+        assert container.state == "running"
+
+    launch_rancher_compose_from_file(
+        client, RCCOMMANDS_SUBDIR, "dc1.yml", env_name,
+        "scale test1=4", "Setting scale", "rc1.yml")
+
+    time.sleep(10)
+    # Confirm service is active and the containers are running
+    service = client.wait_success(service, 300)
+    assert service.state == "active"
```

Comment:
check scale is set explicitly to what is expected and check container state being "running"